packages/store/src/internal/dispatcher.ts

```diff
@@ -49,15 +50,18 @@ export class InternalDispatcher {
     return result.pipe(leaveNgxs(this._ngxsExecutionStrategy));
   }
 
-  private dispatchByEvents(actionOrActions: any | any[]): Observable<any> {
+  private dispatchByEvents(actionOrActions: ActionType | ActionType[]): Observable<any> {
     if (Array.isArray(actionOrActions)) {
-      return forkJoin(actionOrActions.map(a => this.dispatchSingle(a)));
+      const isEmpty: boolean = actionOrActions.length === 0;
+      return isEmpty
```

Comment:
You don't need to check array length and return different observables.

```suggestion
      return forkJoin(actionOrActions.map(a => this.dispatchSingle(a))).pipe(defaultIfEmpty(this._stateStream.getValue()))
```

Reply:
I tried to do this (with using defaultIfEmpty), but then all the threads unsubscribe and the tests fall.

this.dispatch([ new ActionEmptyArray(), new ActionN1(), new ActionN2() ])

In the first case, if our action returns dispatch with empty array, then the rest of the actions will not be executed due to unsubscribing

Reply:
@markwhitfeld what do you think?

Reply:
forkJoin([ ]).pipe( defaultIfEmpty(this._stateStream.getValue())

equals to

EMPTY.pipe( map(() => this._stateStream.getValue()) )

But we need to avoid unsubscribing

Reply:
> forkJoin([ ]).pipe( defaultIfEmpty(this._stateStream.getValue())

> equals to

> EMPTY.pipe( map(() => this._stateStream.getValue()) )

It's not true, run it in the console and you will see the difference.

> all the threads unsubscribe

What threads?

> But we need to avoid unsubscribing

What?

Reply:
Why do you lie? :disappointed: 

![image](https://user-images.githubusercontent.com/7337691/73533063-a8d1f700-4426-11ea-8276-0adac4cde201.png)

Reply:
I think that I prefer @splincode 's code because it makes the scenario that it is handling a bit more explicit. To be clear, I prefer this:
```ts
      const isEmpty: boolean = actionOrActions.length === 0;
      return isEmpty
        ? of(this._stateStream.getValue())
        : forkJoin(actionOrActions.map((action: ActionType) => this.dispatchSingle(action)));
```
It may be more code but I think that it is better to be explicit here. I would never expect `forkJoin` to return an empty observable in any other situation.

Reply:
As you wish.
`isEmpty` with ternary seems to be an overhead for me :smile:

Reply:
@arturovt Maybe a guard clause would be better?
```ts
      if (actionOrActions.length === 0) return of(this._stateStream.getValue());
      return forkJoin(actionOrActions.map((action: ActionType) => this.dispatchSingle(action)));
```

Reply:
It seems to me that such code is really better if the tests pass

```ts
forkJoin(actionOrActions.map((action: ActionType) => this.dispatchSingle(action))).pipe(
   defaultIfEmpty(this._stateStream.getValue()) // returning when forkJoin([ ])
);
```

Reply:
I apparently didn’t check the code and Arthur was right

Reply:
Thank you for checking, Max!